tests/test_api_callable.py

```diff
@@ -141,6 +141,27 @@ def test_call_kwargs(self):
         self.assertEqual(my_callable(None, CallOptions(key='updated')),
                          'updated')
 
+    def test_call_merge_options_metadata(self):
+        settings_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'val1'), ('key2', 'val2')]
+        }
+
+        settings = _CallSettings(kwargs=settings_kwargs)
+        my_callable = api_callable.create_api_call(
+            lambda _req, _timeout, **kwargs: kwargs, settings)
+
+        expected_kwargs = settings_kwargs
+        self.assertEqual(my_callable(None), expected_kwargs)
+
+        expected_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'updated_val1'), ('key2', 'val2')]
+        }
+        self.assertEqual(
+            my_callable(None, CallOptions(metadata=[('key1', 'updated_val1')])),
```

Comment:
I don't know the details of gax, but what I meant here was the ordering of items in an **array**, which is an element inside a dict (kwargs). This is the key point of this PR: merging not the dict itself (it is already done in merge() method), but deeply merging **only one** specific item in the dict, which is an array of tuples (so order is preserved), and accessed by the key "metadata".